context/src/junit/bindings.rs

```diff
@@ -31,6 +32,193 @@ pub struct BindingsReport {
+                let timestamp_micros = testcases
+                    .iter()
+                    .map(|tc| tc.timestamp_micros.unwrap_or(0))
+                    .max();
+                let time = testcases.iter().map(|tc| tc.time.unwrap_or(0.0)).sum();
+                BindingsTestSuite {
+                    name,
+                    tests,
+                    disabled,
+                    errors: 0,
+                    failures,
+                    timestamp,
+                    timestamp_micros,
+                    time: Some(time),
+                    test_cases: testcases,
+                    properties: vec![],
+                    system_out: None,
+                    system_err: None,
+                    extra: HashMap::new(),
+                }
+            })
+            .collect();
+        let (report_time, report_failures, report_tests) =
+            test_suites.iter().fold((0.0, 0, 0), |acc, ts| {
+                (
+                    acc.0 + ts.time.unwrap_or(0.0),
+                    acc.1 + ts.failures,
+                    acc.2 + ts.tests,
+                )
+            });
+        let (name, timestamp, timestamp_micros) = match uploader_metadata {
+            Some(t) => (
+                t.origin,
+                Some(t.upload_time.unwrap_or_default().seconds),
+                Some(
+                    chrono::Duration::nanoseconds(t.upload_time.unwrap_or_default().nanos as i64)
+                        .num_microseconds()
+                        .unwrap_or_default(),
+                ),
+            ),
+            None => ("Unknown".to_string(), None, None),
+        };
+        BindingsReport {
+            name,
+            test_suites,
+            time: Some(report_time),
+            uuid: None,
+            timestamp,
+            timestamp_micros,
+            errors: 0,
+            failures: report_failures,
+            tests: report_tests,
+        }
+    }
+}
+
+impl From<TestCaseRun> for BindingsTestCase {
+    fn from(
+        TestCaseRun {
+            name,
+            parent_name,
+            classname,
+            started_at,
+            finished_at,
+            status,
+            status_output_message,
+            id,
+            file,
+            line,
+            attempt_number,
+        }: TestCaseRun,
+    ) -> Self {
+        let (timestamp, timestamp_micros) = match started_at {
+            Some(started_at) => (
+                Some(started_at.seconds),
+                Some(started_at.nanos as i64 * 1000),
+            ),
+            None => (None, None),
+        };
+        let time = match (started_at, finished_at) {
+            (Some(started_at), Some(finished_at)) => Some(
+                (finished_at.seconds - started_at.seconds) as f64
+                    + (finished_at.nanos - started_at.nanos) as f64 / 1_000_000_000.0,
+            ),
+            _ => None,
+        };
```

Comment:
nit: `chrono` can be used for this too. Any reason why not to use `prost-wkt` / `prost-wkt-types` to convert proto `Timestamp` to `chrono` for querying and manipulation? Seems like it would make a lot of these operations much easier:
https://docs.rs/prost-wkt-types/latest/prost_wkt_types/struct.Timestamp.html#method.from-2